qbraid_qir/qasm3/oq3_maps.py

```diff
@@ -40,6 +42,17 @@
 
 
 def qasm3_expression_op_map(op_name: str, left, right):
+    """
+    Return the result of applying the given operator to the given operands.
+
+    Args:
+        op_name (str): The operator name.
+        left: The left operand.
+        right: The right operand.
+
+    Returns:
+        The result of applying the operator to the operands.
+    """
     try:
         return OPERATOR_MAP[op_name](left, right)
     except KeyError as exc:
```

Comment:
Could this be rewritten with generic variable number of args? e.g.

```python
def qasm3_expression_op_map(op_name: str, *args):
    ...
    
    try:
        return OPERATOR_MAP[op_name](*args)
    except KeyError as exc:
```

If not, or if it's important to require exactly two args here to pass through, could you maybe describe a little more about what these "operands" are? Perhaps type hints could help and/or including the types in the docstring.

Reply:
qasm3 only supports unary or binary ops. This method specifically handles binary ops with `left` and `right` being numeric values (int or float). While there is support for imaginary and angle types, `pyqir` does not support them (as far as I know right now) 

I'll rename the method and provide type hints here.